pkgs/by-name/qq/qqmusic/package.nix

```
@@ -0,0 +1,140 @@
+{
+  fetchurl,
+  stdenv,
+  autoPatchelfHook,
+  makeWrapper,
+  lib,
+  makeDesktopItem,
+  copyDesktopItems,
+  # QQ Music dependencies
+  alsa-lib,
+  at-spi2-atk,
+  at-spi2-core,
+  cairo,
+  cups,
+  dbus,
+  expat,
+  gdk-pixbuf,
+  glib,
+  gtk3,
+  libdbusmenu,
+  libglvnd,
+  libpulseaudio,
+  mesa,
+  nspr,
+  nss,
+  pango,
+  pciutils,
+  udev,
+  xorg,
+  ...
+}:
+################################################################################
+# Mostly based on qqmusic-bin package from AUR:
+# https://aur.archlinux.org/packages/qqmusic-bin
+################################################################################
+let
+  libraries = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    cairo
+    cups
+    dbus
+    expat
+    gdk-pixbuf
+    glib
+    gtk3
+    libdbusmenu
+    libglvnd
+    libpulseaudio
+    mesa
+    nspr
+    nss
+    pango
+    pciutils
+    udev
+    xorg.libX11
+    xorg.libxcb
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXScrnSaver
+    xorg.libXtst
+  ];
+in
+stdenv.mkDerivation {
+  pname = "qqmusic";
+  version = "1.1.5";
+  src = fetchurl {
+    url = "https://dldir1.qq.com/music/clntupate/linux/deb/qqmusic_1.1.5_amd64_.deb";
```

Consider referring to the package version in the download link, via `finalAttrs`
```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "qqmusic";
  version = "1.1.5";
  src = fetchurl {
    url = "https://dldir1.qq.com/music/clntupate/linux/deb/qqmusic_${finalAttrs.version}_amd64_.deb";
```